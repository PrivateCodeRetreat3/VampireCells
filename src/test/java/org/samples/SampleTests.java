package org.samples;


import com.spun.util.ArrayUtils;
import com.spun.util.NumberUtils;
import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;
import org.lambda.utils.Grid;

import java.util.ArrayList;
import java.util.List;

public class SampleTests
{
//  @Test
//  public void testNormalCellDiesWhenLonely()
//  {
//    final NormalCell[] possible = new NormalCell[] {new NormalCell(1), new NormalCell(0), new NormalCell(99), new VampireCell() };
//    CombinationApprovals.verifyAllCombinations(this::getNext, possible, possible, possible, possible, possible, possible, possible, possible, possible);
//  }

  @Test
  public void testNeighborScenarios()
  {
    final NormalCell[] possible = new NormalCell[] {new NormalCell(1), new NormalCell(0), new NormalCell(99), new VampireCell() };
    CombinationApprovals.verifyAllCombinations(this::getNext, possible, new Integer[]{0, 1, 2, 3, 4}, new Integer[]{0, 1, 2, 3});
  }
  private String getNext(NormalCell center, NormalCell topLeft, NormalCell top, NormalCell topRight, NormalCell left,
                         NormalCell right, NormalCell bottomLeft, NormalCell bottom, NormalCell bottomRight) {
    final NormalCell next = center.next(List.of(topLeft, topRight, left, top, right, bottomLeft, bottom, bottomRight));
    return next.toString();
  }
  private String getNext(NormalCell center, Integer aliveNeighborCount, Integer vampireNeihborCount) {
    int deadNeighborCount = 8 - aliveNeighborCount - vampireNeihborCount;
    List<NormalCell> neighbors = new ArrayList<>();
    for (int i = 0; i < aliveNeighborCount; i++) {
      neighbors.add(new NormalCell(1));
    }
    for (int i = 0; i < vampireNeihborCount; i++) {
      neighbors.add(new VampireCell());
    }
    for (int i = 0; i < deadNeighborCount; i++) {
      neighbors.add(new NormalCell(0));
    }
    neighbors = List.of(NumberUtils.getShuffled(neighbors.toArray(new NormalCell[8]), 8));
    return center.next(neighbors).toString();
  }

}
