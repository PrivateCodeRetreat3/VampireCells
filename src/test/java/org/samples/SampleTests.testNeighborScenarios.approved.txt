[01, 0, 0] => __ 
[01, 0, 1] => V  
[01, 0, 2] => V  
[01, 0, 3] => V  
[01, 1, 0] => __ 
[01, 1, 1] => V  
[01, 1, 2] => V  
[01, 1, 3] => V  
[01, 2, 0] => 02 
[01, 2, 1] => 02 
[01, 2, 2] => 02 
[01, 2, 3] => 02 
[01, 3, 0] => 02 
[01, 3, 1] => 02 
[01, 3, 2] => 02 
[01, 3, 3] => 02 
[01, 4, 0] => __ 
[01, 4, 1] => V  
[01, 4, 2] => V  
[01, 4, 3] => V  
[__, 0, 0] => __ 
[__, 0, 1] => __ 
[__, 0, 2] => __ 
[__, 0, 3] => __ 
[__, 1, 0] => __ 
[__, 1, 1] => __ 
[__, 1, 2] => __ 
[__, 1, 3] => __ 
[__, 2, 0] => __ 
[__, 2, 1] => __ 
[__, 2, 2] => __ 
[__, 2, 3] => __ 
[__, 3, 0] => 01 
[__, 3, 1] => 01 
[__, 3, 2] => 01 
[__, 3, 3] => 01 
[__, 4, 0] => __ 
[__, 4, 1] => __ 
[__, 4, 2] => __ 
[__, 4, 3] => __ 
[99, 0, 0] => __ 
[99, 0, 1] => V  
[99, 0, 2] => V  
[99, 0, 3] => V  
[99, 1, 0] => __ 
[99, 1, 1] => V  
[99, 1, 2] => V  
[99, 1, 3] => V  
[99, 2, 0] => V  
[99, 2, 1] => V  
[99, 2, 2] => V  
[99, 2, 3] => V  
[99, 3, 0] => V  
[99, 3, 1] => V  
[99, 3, 2] => V  
[99, 3, 3] => V  
[99, 4, 0] => __ 
[99, 4, 1] => V  
[99, 4, 2] => V  
[99, 4, 3] => V  
[V , 0, 0] => V  
[V , 0, 1] => V  
[V , 0, 2] => __ 
[V , 0, 3] => __ 
[V , 1, 0] => V  
[V , 1, 1] => V  
[V , 1, 2] => __ 
[V , 1, 3] => __ 
[V , 2, 0] => V  
[V , 2, 1] => V  
[V , 2, 2] => __ 
[V , 2, 3] => __ 
[V , 3, 0] => V  
[V , 3, 1] => V  
[V , 3, 2] => __ 
[V , 3, 3] => __ 
[V , 4, 0] => V  
[V , 4, 1] => V  
[V , 4, 2] => __ 
[V , 4, 3] => __ 
